fuel-tx/src/transaction/consensus_parameters.rs

```diff
@@ -15,21 +15,14 @@ pub use gas::{
 };
 
 const MAX_GAS: u64 = 100_000_000;
-const MAX_SIZE: u64 = 17 * 1024 * 1024;
+const MAX_SIZE: u64 = 110 * 1024;
 
-/// A collection of parameters for convenience
+/// A versioned set of consensus parameters.
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-#[cfg_attr(feature = "serde", serde(default))]
-pub struct ConsensusParameters {
-    pub tx_params: TxParameters,
-    pub predicate_params: PredicateParameters,
-    pub script_params: ScriptParameters,
-    pub contract_params: ContractParameters,
-    pub fee_params: FeeParameters,
-    pub chain_id: ChainId,
-    pub gas_costs: GasCosts,
-    pub base_asset_id: AssetId,
+pub enum ConsensusParameters {
+    /// Version 1 of the consensus parameters
+    V1(ConsensusParametersV1),
```

Comment:
Should we consider `#[non_exhaustive]` here?

Reply:
No, since serve will handle it=)